llvm/lib/Transforms/Vectorize/VPlanRecipes.cpp

```diff
@@ -768,6 +768,27 @@ void VPInstruction::execute(VPTransformState &State) {
             /*IsScalar*/ GeneratesPerFirstLaneOnly);
 }
 
+InstructionCost VPInstruction::computeCost(ElementCount VF,
+                                           VPCostContext &Ctx) const {
+  if (Instruction::isBinaryOp(getOpcode())) {
+    if (!getUnderlyingValue())
```

Comment:
Can you add a TODO here to deal with this case? Even if there isn't an underlying value, the cost of an add is still not zero. Alternatively, if there should always be an underlying value you could assert this?

Reply:
Yep, added a TODO,thanks!

 I don't think it can be an assert, as there are cases where we create binary ops without underlying instructions in some cases.